azavea/tilejson.io
# Add layer boxes to sidebar

## Overview

Layer boxes replace the huge TileJSON teat area on the side bar. These boxes show layer names and the order of the layers. They provide the ability to click in and learn details of the layer. They also provide the ability to look at the TileJSON for the individual layer. A button is available to delete a specific layer or to edit the layer information. Lastly, a quick add box allows the user to throw in a tile url and hit enter to see it as a layer. (Also, hitting enter works on forms now too!)

Connects #8 

### Demo

Starting screen
<img width="1280" alt="screen shot 2017-10-24 at 8 39 07 am" src="https://user-images.githubusercontent.com/3959096/31943579-213a42ec-b897-11e7-9ccd-90d7d7fe0a5a.png">

Enter tile url into quick add
<img width="1280" alt="screen shot 2017-10-24 at 8 39 19 am" src="https://user-images.githubusercontent.com/3959096/31943610-351aa914-b897-11e7-8c1b-6d4b142c47cd.png">

See layer
<img width="1280" alt="screen shot 2017-10-24 at 8 39 34 am" src="https://user-images.githubusercontent.com/3959096/31943626-43c57c78-b897-11e7-973f-a50a5b42d08a.png">

View layer details
<img width="1280" alt="screen shot 2017-10-24 at 8 39 39 am" src="https://user-images.githubusercontent.com/3959096/31943692-7fef04d0-b897-11e7-84a6-aa7ba37bf1c0.png">

View layer source
<img width="1280" alt="screen shot 2017-10-24 at 8 39 41 am" src="https://user-images.githubusercontent.com/3959096/31943703-877a4d90-b897-11e7-881b-180bd079ff04.png">

Add new layer
<img width="1280" alt="screen shot 2017-10-24 at 8 39 54 am" src="https://user-images.githubusercontent.com/3959096/31943710-901887aa-b897-11e7-9218-4168997219c2.png">

Edit popup for first layer
<img width="1280" alt="screen shot 2017-10-24 at 8 40 11 am" src="https://user-images.githubusercontent.com/3959096/31943713-948f0bec-b897-11e7-945a-6cd901571767.png">

Change name and tile url
<img width="1280" alt="screen shot 2017-10-24 at 8 40 33 am" src="https://user-images.githubusercontent.com/3959096/31943715-995b0ffe-b897-11e7-90e9-05fc4f5a2616.png">

See the change in place 
<img width="1280" alt="screen shot 2017-10-24 at 8 40 53 am" src="https://user-images.githubusercontent.com/3959096/31943720-9f417ee4-b897-11e7-9b1d-197cf38526a4.png">

Delete old layer
<img width="1280" alt="screen shot 2017-10-24 at 8 40 59 am" src="https://user-images.githubusercontent.com/3959096/31943724-a39c52fc-b897-11e7-94af-c91bacb80b7a.png">


## Testing Instructions

 * Pull branch, serve locally
 * Test the Quick Add box by adding some tile url to that. Also add using the add popup. 
 * Test the view details expand/collapse button for each layer.
 * Test the view code button for each layer.
 * Test the delete button for the layers.
 * Test editing a layer and replacing the url with a new one and see the layer change while keeping its place.
